hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzoneClientUtils.java

```diff
@@ -64,4 +73,78 @@ public static BucketLayout resolveLinkBucketLayout(OzoneBucket bucket,
     }
     return bucket.getBucketLayout();
   }
+
+  /**
+   * This API used to resolve the client side configuration preference for file
+   * system layer implementations.
+   *
+   * @param replication                - replication value passed from FS API.
+   * @param clientConfiguredReplConfig - Client side configured replication
+   *                                   config.
+   * @param bucketReplConfig           - server side bucket default replication
+   *                                  config.
+   * @param config                     - Ozone configuration object.
+   * @return client resolved replication config.
+   */
+  public static ReplicationConfig resolveClientSideReplicationConfig(
```

Comment:
Should we add a unit test for this logic? There is a fair bit going on here to not have a test covering it.

Reply:
Added few tests specific to this API